corehq/apps/domain/views/settings.py

```diff
@@ -517,3 +520,87 @@ class ManageDomainMobileWorkersView(ManageMobileWorkersMixin, BaseAdminProjectSe
     page_title = gettext_lazy("Manage Mobile Workers")
     template_name = 'enterprise/manage_mobile_workers.html'
     urlname = 'domain_manage_mobile_workers'
+
+
+@method_decorator(toggles.CUSTOM_DOMAIN_BANNER_ALERTS.required_decorator(), name='dispatch')
+class ManageDomainAlertsView(BaseAdminProjectSettingsView):
+    template_name = 'domain/admin/manage_alerts.html'
+    urlname = 'domain_manage_alerts'
+    page_title = gettext_lazy("Manage Project Alerts")
+
+    @property
+    def page_context(self):
+        return {
+            'form': self.form,
+            'alerts': [
+                {
+                    'active': alert.active,
+                    'html': alert.html,
+                    'id': alert.id,
+                    'created_by_user': alert.created_by_user,
+                }
+                for alert in CommCareHQAlert.objects.filter(created_by_domain=self.domain)
+            ]
+        }
+
+    @cached_property
+    def form(self):
+        if self.request.method == 'POST':
+            return DomainAlertForm(self.request.POST)
+        return DomainAlertForm()
+
+    def post(self, request, *args, **kwargs):
+        command = request.POST.get('command')
+        if command:
+            self._apply_command(request, command)
+        elif self.form.is_valid():
+            self._create_alert()
+            messages.success(request, _("Alert saved!"))
+        else:
+            messages.error(request, _("There was an error saving your alert. Please try again!"))
```

Comment:
Will additional details about the error(s) be displayed in addition to this message? Simply trying again without changing the submitted values in some way is likely to produce the same error.

The same applies below for unsuccessful `_update_alert`

Reply:
this last else block is just in case someone does something funky on the page or for rare unexpected scenarios.
If the forms are used as intended, there will either be a "command" (activate/deactivate) or the form with the text field (which is a required field on the form). Errors are really not expected on this page as for how it is now.
Hence the generic message. Open to suggestions if I am being too positive.

Reply:
The typical purpose of form validation is to construct a detailed list of errors to be displayed to the user so they can be corrected. This form does not seem to be following that pattern. Since this would only happen if "someone does something funky on the page or for rare unexpected scenarios", I suggest simply raising an exception so we can find the issue in Sentry and have more information for debugging.
```suggestion
            raise ValidationError(form.errors.as_text())
```

Reply:
> The typical purpose of form validation is to construct a detailed list of errors to be displayed to the user so they can be corrected

Right. Not sure if I misguided earlier. The error messages related to the form will be shown within the form itself. This error message is visible on the top as a generic message. Here is how it will look like.

![Screenshot from 2023-10-11 14-42-03](https://github.com/dimagi/commcare-hq/assets/3864163/4ec2dbc2-965d-4c76-a97c-a4eae539c3c1)

Reply:
Great. My original question here was if specific validation errors are displayed for each field:

> Will additional details about the error(s) be displayed in addition to this message?

I now see the answer is "Yes." Thanks for the screenshot. No change is required here.

Reply:
Cool, sorry about the confusion.